ui/vr_webgl.js

```diff
@@ -0,0 +1,611 @@
+    // to not mantain a value over 90º or under -90º.
+      this.positionY_ -= pitch;
+    }
+
+    const out2 = shaka.ui.Matrix4x4.create();
+    shaka.ui.Matrix4x4.rotateZ(out2, shaka.ui.Matrix4x4.create(), roll);
+
+    // Rotate local axis
+    shaka.ui.Matrix4x4.multiply(matrix, out2, matrix);
+
+    this.renderGL_(false);
+  }
+
+  /**
+   * @param {number} amount
+   */
+  zoom(amount) {
+    const zoomMin = 20;
+    const zoomMax = 100;
+    amount /= 50;
+    if (this.fieldOfView_ >= zoomMin && this.fieldOfView_ <= zoomMax) {
+      this.fieldOfView_ += amount;
+    }
+    if (this.fieldOfView_ < zoomMin) {
+      this.fieldOfView_ = zoomMin;
+    } else if (this.fieldOfView_ > zoomMax) {
+      this.fieldOfView_ = zoomMax;
+    }
+    this.renderGL_(false);
+  }
+
+  /**
+   * @return {number}
+   */
+  getFieldView() {
+    return this.fieldOfView_;
+  }
+
+  /**
+   * @param {number} fieldView
+   */
+  setFieldView(fieldView) {
+    this.fieldOfView_ = fieldView;
+    this.renderGL_(false);
+  }
+
+  /**
+   * @return {number}
+   */
+  getNorth() {
+    shaka.ui.Matrix4x4.getRotation(this.currentQuaternion_, this.viewMatrix_);
+
+    const angles = this.toEulerAngles_(this.currentQuaternion_);
+
+    let result = ((angles.yaw * 180) / Math.PI);
+    if (result < 0) {
+      result = 360 + result;
+    }
+    return result;
+  }
+
+  /**
+   * @param {boolean=} firstTime
+   */
+  reset(firstTime = true) {
+    const steps = 20;
+
+    if (firstTime) {
+      shaka.ui.Matrix4x4.getRotation(
+          this.currentQuaternion_, this.viewMatrix_);
+      this.cont_ = 0;
+      this.diff_ = shaka.ui.MatrixQuaternion.create();
+      this.diff_[0] =
+          (this.currentQuaternion_[0] - this.originalQuaternion_[0]) / steps;
+      this.diff_[1] =
+          (this.currentQuaternion_[1] - this.originalQuaternion_[1]) / steps;
+      this.diff_[2] =
+          (this.currentQuaternion_[2] - this.originalQuaternion_[2]) / steps;
+      this.diff_[3] =
+          (this.currentQuaternion_[3] - this.originalQuaternion_[3]) / steps;
+    }
+
+    this.currentQuaternion_[0] -= this.diff_[0];
+    this.currentQuaternion_[1] -= this.diff_[1];
+    this.currentQuaternion_[2] -= this.diff_[2];
+    this.currentQuaternion_[3] -= this.diff_[3];
+
+    // Set the view to the original matrix
+    const out = shaka.ui.Matrix4x4.create();
+
+    shaka.ui.MatrixQuaternion.normalize(
+        this.currentQuaternion_, this.currentQuaternion_);
+
+    shaka.ui.Matrix4x4.fromQuat(out, this.currentQuaternion_);
+
+    this.viewMatrix_ = out;
+
+    if (this.cont_ < steps) {
+      new shaka.util.Timer(() => {
+        this.reset(false);
+        this.positionY_ = 0;
+        this.cont_++;
+        this.renderGL_(false);
+      }).tickAfter(this.animDuration_ / steps);
+    } else {
+      shaka.ui.Matrix4x4.fromQuat(out, this.originalQuaternion_);
+      this.viewMatrix_ = out;
+      this.renderGL_(false);
```

Comment:
Won't this lead to `renderGL_` being called twice on the last step?
Since it's called here, at the end of the `reset` method, but it's also called after the `reset` method is, inside the timer.